airflow/jobs.py

```diff
@@ -2559,7 +2569,13 @@ def signal_handler(signum, frame):
             while True:
                 # Monitor the task to see if it's done
                 return_code = self.task_runner.return_code()
+
                 if return_code is not None:
+                    if return_code != 0:
+                        msg = ("LocalTaskJob process exited with non zero status "
+                               "{}".format(return_code))
+                        raise AirflowException(msg)
```

Comment:
Adding `raise` here will change the behavior.

Reply:
that's exactly the issue in [AIRFLOW-578](https://issues.apache.org/jira/browse/AIRFLOW-578).
BaseJob ignores the return code of the spawned process. which makes even that process is killed or returned abnormally, it will think it finishes with success

Reply:
Rather than raising an exception here, to then get picked up by the on_failure you added, could we instead just handle it directly here?

Reply:
I think it's better to let the caller to handle it to reusing existing error handling logic , rather than reimplement similar logic here